src/Component/spec/Filter/NumericRangeFilterSpec.php

```diff
@@ -0,0 +1,198 @@
+        );
+    }
+
+    function it_filters_number_to(
+        DataSourceInterface $dataSource,
+        ExpressionBuilderInterface $expressionBuilder,
+    ): void {
+        $dataSource->getExpressionBuilder()->willReturn($expressionBuilder);
+        $expressionBuilder->lessThanOrEqual('number', '8')->willReturn('EXPR');
+
+        $dataSource->getExpressionBuilder()->shouldBeCalledOnce();
+        $dataSource->restrict('EXPR')->shouldBeCalled();
+
+        $this->apply(
+            $dataSource,
+            'number',
+            [
+                'lessThan' => '8',
+            ],
+            [],
+        );
+    }
+
+    function it_filters_number_to_without_inclusive_to(
+        DataSourceInterface $dataSource,
+        ExpressionBuilderInterface $expressionBuilder,
+    ): void {
+        $dataSource->getExpressionBuilder()->willReturn($expressionBuilder);
+        $expressionBuilder->lessThan('number', '9')->willReturn('EXPR');
+
+        $dataSource->getExpressionBuilder()->shouldBeCalledOnce();
+        $dataSource->restrict('EXPR')->shouldBeCalled();
+
+        $this->apply(
+            $dataSource,
+            'number',
+            [
+                'lessThan' => '9',
+            ],
+            [
+                'inclusive_to' => false,
+            ],
+        );
+    }
+
+    function it_filters_money_in_specified_range(
+        DataSourceInterface $dataSource,
+        ExpressionBuilderInterface $expressionBuilder,
+    ): void {
+        $dataSource->getExpressionBuilder()->willReturn($expressionBuilder);
+        $expressionBuilder->greaterThanOrEqual('number', 12)->willReturn('EXPR2');
+        $expressionBuilder->lessThanOrEqual('number', 120)->willReturn('EXPR3');
+
+        $dataSource->getExpressionBuilder()->shouldBeCalledOnce();
+        $dataSource->restrict('EXPR2')->shouldBeCalledOnce();
+        $dataSource->restrict('EXPR3')->shouldBeCalledOnce();
+
+        $this->apply(
+            $dataSource,
+            'number',
+            [
+                'greaterThan' => '12.00',
+                'lessThan' => '120.00',
+            ],
+            [],
+        );
+    }
+
+    function its_amount_scale_and_mode_can_be_configured(
+        DataSourceInterface $dataSource,
+        ExpressionBuilderInterface $expressionBuilder,
+    ): void {
+        $dataSource->getExpressionBuilder()->willReturn($expressionBuilder);
+        $expressionBuilder->greaterThanOrEqual('number', 121)->willReturn('EXPR');
+        $expressionBuilder->lessThanOrEqual('number', 259)->willReturn('EXPR1');
+
+        $dataSource->getExpressionBuilder()->shouldBeCalledOnce();
+        $dataSource->restrict('EXPR')->shouldBeCalledOnce();
+        $dataSource->restrict('EXPR1')->shouldBeCalledOnce();
+
+        $this->apply(
+            $dataSource,
+            'number',
+            [
+                'greaterThan' => '120.78',
+                'lessThan' => '258.51',
+            ],
+            [
+                'scale' => 0,
```

Comment:
Maybe include a test where the scale is not 0, since that's the default value now.